### DISCUSSION TITLE
如何在插件中获取控制台服务来源中配置的来源地址

### DISCUSSION THREAD

**Discussion by titrxw** [higress-group/higress#1489] [Q&A]
## Why do you need it?
 Is your feature request related to a problem? Please describe in details
如何在插件中获取控制台服务来源中配置的来源地址
![QQ_1730977213680](https://github.com/user-attachments/assets/5e811bdd-0b1e-415d-9e4a-1303185d40b2)

## How could it be?
A clear and concise description of what you want to happen. You can explain more about input of the feature, and output of it.

**Comment by johnlanni** [comment]
看下这篇[文章](https://higress.io/docs/latest/user/wasm-go/#%E5%9C%A8%E6%8F%92%E4%BB%B6%E4%B8%AD%E8%AF%B7%E6%B1%82%E5%A4%96%E9%83%A8%E6%9C%8D%E5%8A%A1)，到服务列表看到的服务名称就是这里说的FQDN，用这个方式来生成 client：

```go
  config.client = wrapper.NewClusterClient(wrapper.FQDNCluster{
    FQDN: serviceName,
    Port: servicePort,
   }
```

**Comment by titrxw** [comment]
config.client = wrapper.NewClusterClient(wrapper.FQDNCluster{
    FQDN:"cdn.dns",
    Port: 80,
   }

请求cdn上的一个资源文件， 一直返回404, 实际上这个资源文件是存在的

**Reply by johnlanni** [reply]
发送请求的代码是怎么写的呢？另外cdn可能需要你指定域名，这个cluster初始化可以传入一个 Host 字段来指定域名，或者在 Get 请求的 url 参数里填写带域名的完整 url

**Reply by titrxw** [reply]
有没办法能读取到服务来源中配置的域名地址？

**Reply by johnlanni** [reply]
这个拿不到
